pkg/reconciler/members.go

```diff
@@ -602,6 +595,30 @@ func notifyReadyNodes(
 			for _, member := range ready {
 				go func(m *kdv1.MemberStatus, r *roleInfo) {
 					defer wgReady.Done()
+
+					setupURL, setupURLErr := catalog.AppSetupPackageUrl(cr, r.roleStatus.Name)
+					if setupURLErr != nil {
```

Comment:
since this is a role based property, can we move into the outer role loop

Reply:
and also skip doing looping members if there is no setup url

Reply:
We should still move this check to outer loop no?
Just use otherRole.roleStatus.Name to fetch and check if setup url is present and we can skip the whole iteration

Reply:
The problem was doing so needs a change to the line `wgReady.Add(totalReady)`.

Reply:
I moved the call to get the app setup package out of the member loop. At least that way, we only make the call once per role. 

I don't see a point in further optimizing this function unless I am wrong about the `go func` being inexpensive.